src/oic/oic/provider.py

```diff
@@ -1433,12 +1403,9 @@ def do_client_registration(self, request, client_id, ignore=None):
 
         try:
             self.keyjar.load_keys(request, client_id)
-            try:
-                n_keys = len(self.keyjar[client_id])
-                msg = "found {} keys for client_id={}"
-                logger.debug(msg.format(n_keys, client_id))
-            except KeyError:
-                pass
+            n_keys = len(self.keyjar.get(client_id, []))
+            if n_keys:
+                logger.debug("Found %s keys for client_id=%s", n_keys, client_id)
         except Exception as err:
```

Comment:
Can we be more specific than `Exception` for the  `self.keyjar.load_keys(request, client_id)` failure when we catch?